src/app/core/models/assignee.model.ts

```diff
@@ -0,0 +1,22 @@
+import { Group } from './github/group.interface';
+
+/**
+ * Represents an assignee and its attributes fetched from Github.
+ */
+export class Assignee implements Group {
+  static WithoutAssignee: Assignee = new Assignee({ login: 'Issues/PRs without an assignee', avatar_url: null });
+  login: string;
+  avatarUrl: string;
+
+  constructor(assignee: { login: string; avatar_url: string }) {
+    this.login = assignee.login;
+    this.avatarUrl = assignee.avatar_url;
+  }
+
+  public equals(other: any) {
+    if (!(other instanceof Assignee)) {
+      return false;
+    }
+    return this.login === other.login;
+  }
+}
```

Comment:
I don't see the need for a new class to represent an assignee since an assignee can only be a GitHub User. Unless the assignee has different behaviors or fields from a GitHub user, creating a new class seems unnecessary. Additionally, you created a new service for the assignee, which just parses data from the GitHub user. 

If the concern is about not passing the entire GitHub user object to methods that only require the "login" and "avatarUrl" fields, a simpler solution would be to create an interface called Assignee with these fields and have the GitHub user implement this interface. This way, methods requiring an assignee can use the Assignee type, ensuring only the assignee-specific fields are accessed.

I might be wrong about this, so please feel free to discuss further if you have different considerations.

Reply:
@NereusWB922 okay, after working deeper on the filter PR I think i have a better understanding of the role of each of the models and services.. will try to incorporate the new filter without using a new assignee model

that said, having a new service for assignee is fine right? I'll just rewrite it so that it uses githubUsers instead of the new assignee model class

Reply:
Hi @kokerinks, then in this case, you can just use the method getUsersAssignable from github service right? New service seems unnecessary also.

Reply:
hmm i think it would make the code less easy to understand though. i intentionally designed `assignee.service.ts` to be similar in structure to `milestone.service.ts`, since they both have a similar purpose of handling logic for their respective filters.`assignee.service.ts` currently also "caches" the assignees taken from `getUsersAssignable`, which will help to reduce calls to github APIs.

There are some options i see here:
1. Use `getUsersAssignable` to retrieve each time and without any caching 
  - will cause structure of `assignee` filter implementation to differ from milestone
  - will result in more github api calls
2. Implement caching in `github.service.ts` 
  - code structure would not be consistent with `milestone.service.ts`, where the milestones are "cached" at
3. Implement caching in a new `assignee.service.ts`
  - will result in a new file, but will align better with the current milestone filter implementation in terms of code structure
  
Personally, I think having the new file is necessary.. i understand the potential concern of having too many service files, but maybe we can reorganise the services into new folders to declutter the `services` folder.

bit of a read, let me know what you think!

Reply:
Hi @kokerinks, thanks for the detailed explanation. I can see where you are coming from and agree that having the filtering logic in the GitHub User model would make it too complicated. After reviewing our codebase, I think our current filter service is too complicated and definitely needs some refactoring. For now, you can stick with your implementation. I will create a separate GitHub issue regarding the refactoring of the filter service.

Reply:
Just let me know when this pr is ready for review again. Thanks